openwisp_monitoring/device/base/models.py

```diff
@@ -399,21 +463,8 @@ def threshold_crossed(sender, metric, alert_settings, target, first_time, **kwar
             monitoring = target.monitoring
         except DeviceMonitoring.DoesNotExist:
             monitoring = DeviceMonitoring.objects.create(device=target)
-        status = 'ok' if metric.is_healthy else 'problem'
-        related_status = 'ok'
-        for related_metric in monitoring.related_metrics.filter(is_healthy=False):
-            if monitoring.is_metric_critical(related_metric):
-                related_status = 'critical'
-                break
```

Comment:
if you remember our discussion, we said that we would have to avoid breaking here in order to set the status to critical only when all the critical metrics are unalthy.

Reply:
These are removed lines. The logic is added in `get_monitoring_status()`.